pkg/shp/cmd/build/run.go

```diff
@@ -58,6 +72,46 @@ func (r *RunCommand) Validate() error {
 	return nil
 }
 
+// tailLogs start tailing logs for each container name, if not started already.
+func (r *RunCommand) tailLogs(pod *corev1.Pod) {
+	for _, container := range pod.Spec.Containers {
+		if _, exists := r.tailLogsStarted[container.Name]; exists {
+			continue
+		}
+		r.tailLogsStarted[container.Name] = true
+		r.logTail.Start(pod.GetNamespace(), pod.GetName(), container.Name)
+	}
+}
+
+// onEvent reacts on pod state changes, to start and stop tailing container logs.
+func (r *RunCommand) onEvent(pod *corev1.Pod) error {
+	switch pod.Status.Phase {
+	case corev1.PodPending:
+		fmt.Fprintf(r.ioStreams.Out, "Pod '%s' is pending...\n", pod.GetName())
```

Comment:
the tkn code has some additional logic around issues with pulling images that is appropriate here.... though I suggest making this a default case, to handle any unexpected phases ... you could string(phase) for your message here after this check:

```
	// Handle any issues with pulling images that may fail
	for _, c := range pod.Status.Conditions {
		if c.Type == corev1.PodInitialized || c.Type == corev1.ContainersReady {
			if c.Status == corev1.ConditionUnknown {
				return fmt.Errorf(c.Message)
			}
		}
	}

```

Reply:
That's rather good, thanks for pointing that out!

Reply:
The changes to show the pod is on unknown state is [here](https://github.com/otaviof/cli/blob/a7e0eb8d891c0c1ec211c5a1fa89258b9b36d561/pkg/shp/cmd/build/run.go#L105-L106).

Reply:
ok that's good, but you have still not replicated the conditions interrogation and testing for when to abort based on that

Reply:
it might be easier to discuss it live in the team meeting we are currently on :-)

but while other subjects are getting discussed, I'll give it another go ... giving you more context than my last paste of code  might help

consider 

```
	if pod.Status.Phase == corev1.PodSucceeded ||
		pod.Status.Phase == corev1.PodRunning ||
		pod.Status.Phase == corev1.PodFailed {
		return pod, nil
	}

	// Handle any issues with pulling images that may fail
	for _, c := range pod.Status.Conditions {
		if c.Type == corev1.PodInitialized || c.Type == corev1.ContainersReady {
			if c.Status == corev1.ConditionUnknown {
				return pod, fmt.Errorf(c.Message)
			}
		}
	}
```

that first if statement is analogous to the case stmts you have for those 3 phases

if it is any other phase, then the for loop and condition check come into play

so i.e. for both unknown and pending, or using `default` instead to cover both, you should employ that for loop and return the error if the conditions are met

Reply:
I see, so during `unknown` phase we can surface this error, on the second part of the snippet. Please consider the latest changes.

Reply:
I just refactored this block to check for issues pulling images before the switch-case block, please consider.

Reply:
I may tinker with putting that if in a default clause after we merge this and I update my local clone of the PR branch, but minimally let's go with this for now - thanks